File: hot-framework/src/test/resources/be/solidx/hot/promise/promises3.py
def subThen(result):
    #raise Exception("maerde")
    return result*3

def done(result):
    print result

def then(result):
    return subdeferred.promise()

def fail(e):
    print e

def thenfail(message):
    return message + message

promise = deferred.promise()
promise.then(then,thenfail).then(subThen).done(done).fail(fail)

#subdeferred.resolve(20)
deferred.reject("Merde...")
#deferred.resolve(30)

File: hot-framework/src/test/resources/be/solidx/hot/promise/promises3.js

var promise = deferred.promise()

promise.then(function(result) {
	return subdeferred.promise();
}).then(function(result) {
	return result * 3
},function(value) {
	return value + value;
}).done(function(result) {
	hprint(result)
}).fail(function(message) {
	hprint(message)
})

deferred.reject("Oops")
subdeferred.resolve(30)